pkg/monitoring/metrics/forklift-controller/plan_metrics.go

```diff
@@ -0,0 +1,100 @@
+package forklift_controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var processedPlans = make(map[string]struct{})
```

Comment:
can we act like the plans also in the migrations?
or you didn't do that as you check if we need both migrations and plan?

Reply:
currently we don't need, because the migration is set with atomic counters that always reflect the status, it set the real value each time instead of increasing it, but the plan is have to many combinations and options now that we have 4 labels on the same metric, so we will need to have all the combinations of `Local/Remote.Warm/Cold.Status.ProviderType` which leads to to many counters and if conditions, so this implementation simplifies things.

also I've comment above  `I'ts the migration entity that corresponds to plan, I left it because the original team had it, but do we really need both? I think one of them should be enough but which one, plans or migrations would be better?` so the question is if we really need it and if we do should we add additional labels there as well?